src/service/resources-impl.js

```diff
@@ -1741,28 +1816,57 @@ export class Resource {
   }
 
   /**
-   * Calls element's documentInactiveCallback callback and resets state for
+   * Calls element's unlayoutCallback callback and resets state for
    * relayout in case document becomes active again.
    */
-  documentBecameInactive() {
-    if (this.state_ == ResourceState_.NOT_BUILT) {
+  unlayout() {
+    if (this.state_ != ResourceState_.LAYOUT_COMPLETE) {
```

Comment:
> This is not entirely correct. setInViewport is called regardless of the layout state.

Fixed.

> Also, there could be a risk condition where the state is READY_FOR_LAYOUT or LAYOUT_SCHEDULED.

Should we just set `state_` to `NOT_LAID_OUT` without any `#unlayoutCallback` in this case?